LPS-160210 Add WebDAV URL to Info Panel

beltranrengifo opened the pull request:
[Jira ticket](https://issues.liferay.com/browse/LPS-160210)

## Motivation

Add WebDAV URL to the info panel in Content Dashboard.

## Proposed Solution

Send URL fields in the SpecificFields section of the infoPanel endpoint:

```
"specificFields": {
    "extension": {
      "title": "Extension",
      "type": "String",
      "value": "png"
    },
    "size": {
      "title": "Size",
      "type": "String",
      "value": "359 KB"
    },
    "web-dav-url": {
      "title": "WebDAV URL",
      "type": "URL",
      "value": "http://localhost:8080/webdav/guest/document_library/tree.png"
    },
    "latest-version-url": {
      "title": "Latest Version URL",
      "type": "URL",
      "value": "http://localhost:8080/documents/d/guest/tree-png"
    }
  }
```

Support URL fields in the Content Dashboard Info Panel Section

## How to

- Upload some files to the DL
- Navigate to the CD
- Open the info panel
- Check for the components in the details section

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

beltranrengifo commented:
Hi @cgoncas !
I tried to modify the needed parts in the backend. Not sure if I'm doing it right.
Let me know your thoughts whenever you can.
Thxs!

cgoncas commented on modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/FileEntryContentDashboardItem.java:
```diff
@@ -296,6 +299,8 @@ public String getScopeName(Locale locale) {
 	@Override
 	public Map<String, Object> getSpecificInformation(Locale locale) {
 		return HashMapBuilder.<String, Object>put(
+			"clipboard", _getClipboard()
```
Hey @beltranrengifo !!

We should change it, we should add the webDAV URL as a field and the new latest version URL, we should erase every mention to clipboard.

cgoncas replied:
You should change the frontend as we mention in the daily meeting, so we can read URLs in the frontend side.

And delete the latest URL that now is being shown in the info panel.

Once you finish the frontend changes, please, start with the versions, so we can do as much frontend changes as possible before your PTO next week.

cgoncas replied:
The frontend is what I need from you, because we don't have any frontend in the next week.

In the backend side, we have plenty of hands.

thanks!!

beltranrengifo replied:
Yeah, you're right, thought it was faster that it is. Sorry for the confusion

cgoncas commented:
Hey @beltranrengifo !!

Sorry I have add the backend code with the new information, so you can write the frontend part.

I need from you to adapt the URL section, so they can be copied, currently, they look like this:

![Captura de pantalla 2022-08-17 a las 14 18 48](https://user-images.githubusercontent.com/5776822/185144274-e5e8b626-c613-4a40-96d5-01360c6ffb69.png)


Thanks!!

beltranrengifo commented:
> Hey @beltranrengifo !!
> 
> Sorry I have add the backend code with the new information, so you can write the frontend part.
> 
> I need from you to adapt the URL section, so they can be copied, currently, they look like this:
> 
> ![Captura de pantalla 2022-08-17 a las 14 18 48](https://user-images.githubusercontent.com/5776822/185144274-e5e8b626-c613-4a40-96d5-01360c6ffb69.png)
> 
> Thanks!!

Got it! Thanks

beltranrengifo commented on modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/item/FileEntryContentDashboardItem.java:
```diff
@@ -297,8 +302,12 @@ public String getScopeName(Locale locale) {
 	public Map<String, Object> getSpecificInformation(Locale locale) {
 		return HashMapBuilder.<String, Object>put(
 			"extension", _getExtension()
+		).put(
```
@cgoncas I wonder if we can reorder this object to match the UI.
If not, I could force it in the FE but IDK if it makes a lot of sense

cgoncas replied:
Yeahh!! I will sort it in the backend.

beltranrengifo commented:
@cgoncas I've rebased, added the FE and force pushed.
Just the reorder is needed, I think.
![image](https://user-images.githubusercontent.com/19485114/185358299-a59cfaf0-a792-4065-8705-7a0ede73194b.png)

cgoncas commented:
> @cgoncas I've rebased, added the FE and force pushed. Just the reorder is needed, I think. ![image](https://user-images.githubusercontent.com/19485114/185358299-a59cfaf0-a792-4065-8705-7a0ede73194b.png)

Ohhh @beltranrengifo !!

This is sooo beautiful!!!

cgoncas commented:
I accidentally delete the branch :(, this PR continues in https://github.com/liferay-tango/liferay-portal/pull/2713
